packages/next/src/build/next-config-ts/transpile-config.ts

```
@@ -73,34 +74,57 @@ async function verifyTypeScriptSetup(cwd: string, configFileName: string) {
+  configFileName: string
+): Promise<CompilerOptions> {
+  try {
+    // Ensure TypeScript is installed to use the TypeScript API.
+    await verifyTypeScriptSetup(cwd, configFileName)
 
-  // NOTE: This doesn't fully cover the edge case for setting
-  // "typescript.tsconfigPath" in next config which is currently
-  // a restriction.
-  const tsConfigPath = ts.findConfigFile(
-    cwd,
-    ts.sys.fileExists,
-    'tsconfig.json'
-  )
+    const ts: typeof import('typescript') = require(
+      require.resolve('typescript', { paths: [cwd] })
+    )
 
-  if (!tsConfigPath) {
-    // It is ok to not return ts.getDefaultCompilerOptions() because
-    // we are only looking for paths and baseUrl from tsConfig.
-    return {}
-  }
+    // NOTE: This doesn't fully cover the edge case for setting
+    // "typescript.tsconfigPath" in next config which is currently
+    // a restriction.
+    const tsConfigPath = ts.findConfigFile(
+      cwd,
+      ts.sys.fileExists,
+      'tsconfig.json'
+    )
+
+    if (!tsConfigPath) {
+      // It is ok to not return ts.getDefaultCompilerOptions() because
+      // we are only looking for paths and baseUrl from tsConfig.
+      return {}
+    }
 
-  const configFile = ts.readConfigFile(tsConfigPath, ts.sys.readFile)
-  const parsedCommandLine = ts.parseJsonConfigFileContent(
-    configFile.config,
-    ts.sys,
-    cwd
-  )
+    const configFile = ts.readConfigFile(tsConfigPath, ts.sys.readFile)
+    const parsedCommandLine = ts.parseJsonConfigFileContent(
+      configFile.config,
+      ts.sys,
+      cwd
+    )
 
-  return parsedCommandLine.options
+    return parsedCommandLine.options
+  } catch (cause1) {
+    try {
+      warn(
+        `Failed to parse "${configFileName}" with TypeScript API. Falling back to JSON parsing.`,
+        { cause: cause1 }
+      )
+      // Fallback to JSON parsing. This won't resolve options like "extends".
+      return parseJsonFile(resolve(cwd, 'tsconfig.json')).compilerOptions || {}
```

This looks incorrect, if TS setup or ts config parsing bail out, this try-catch will hide the error and let `parseJsonFile` return a invalid config. I don't think we should do this if there's invalid error provided

---

I think it's fine because the goal here is to get `paths` and `baseUrl`. The strict `tsconfig.json` validation will be handled in a later process, e.g., `loadJsConfig()`, after reading nextConfig `typescript` value like `typescript.tsconfigPath`.

To prevent hiding, it warns instead.

---

> I think it's fine because the goal here is to get paths and baseUrl

With `JSON.parse` you won't know if that's the correct value though since it could come from `extends`.

---

Aligned early fail is mature. Thanks both.